staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -391,6 +392,20 @@ func (t *tracker) add(gvr schema.GroupVersionResource, obj runtime.Object, ns st
 		return err
 	}
 
+	// Only generate the name for objects that are Created
```

Comment:
hmm... I'm not really a fan of replicating server logic/mutations in client-go in inaccurate ways (name generation is resource-specific)... are we also doing things like populating uid and creationTimestamp on creation?

Reply:
> are we also doing things like populating uid and creationTimestamp on creation?

nope,  that parted surprised me too

> name generation is resource-specific

ah, right, this depends on the strategy